### DISCUSSION TITLE
Extract mentions from JSONContent

### DISCUSSION THREAD

**Discussion by migalmeida** [ueberdosis/tiptap#2569] [General Questions] [answered]
Hi! 

First of all, congratulations on the project. It is really easy to use and extend.

I'm using `@tiptap/extension-mention` to implement @ mentions. I was wondering if there's any conventional way to transverse JSONContent to extract all the correspondent `mention` nodes? Either in the extension itself or via Editor API?

Thanks in advance.

**Comment by rfgamaral** [comment, marked as answer]
Hello @migalmeida 👋

Fancy seeing you here, how have you been? 😊

To answer your question, we've done this ourselves pretty recently, here's what we ended up with:

```ts
function getAllNodesAttributesByType(
    doc: ProseMirrorNode,
    nodeType: string,
): Array<NodeAttributes> {
    const result: Array<NodeAttributes> = []

    doc.descendants((node) => {
        if (node.type.name === nodeType) {
            result.push(node.attrs)
        }
    })

    return result
}
```

And for the official `@tiptap/extension-mention` you'd call it something like this:

```ts
getAllNodesAttributesByType(editor.state.doc, 'mention')
```

Please note that the above function **extracts the node attributes** not the node itself, but you can easily adapt it to return what you need.

Hope this helps 👍

**Reply by migalmeida** [reply]
Hey @rfgamaral ! 

Long time no see. Hope you are doing well. 😄 

Thanks for the help, we didn't know that `descendants` method. It is perfect for what we want 🙏
